edit-post/store/effects.js

```diff
@@ -65,23 +66,35 @@ const effects = {
 		// If we do not provide this data the post will be overriden with the default values.
 		const post = select( 'core/editor' ).getCurrentPost( state );
 		const additionalData = [
-			post.comment_status && `comment_status=${ post.comment_status }`,
-			post.ping_status && `ping_status=${ post.ping_status }`,
-			`post_author=${ post.author }`,
+			post.comment_status ? [ 'comment_status', post.comment_status ] : false,
+			post.ping_status ? [ 'ping_status', post.ping_status ] : false,
+			[ 'post_author', post.author ],
 		].filter( Boolean );
 
-		// To save the metaboxes, we serialize each one of the location forms and combine them
-		// We also add the "common" hidden fields from the base .metabox-base-form
-		const formData = values( dataPerLocation )
-			.concat( jQuery( '.metabox-base-form' ).serialize() )
-			.concat( additionalData )
-			.join( '&' );
+		// We gather all the metaboxes locations data and the base form data
+		const baseFormData = new window.FormData( document.querySelector( '.metabox-base-form' ) );
```

Comment:
Nice, we have a few places where we use `jQuery( selector ).serialize()`. It would be awesome if we could get rid of it completely.

Reply:
Yep but not possible at the moment because we need a serialized version of the content to check for changes.